x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ApiKeyFieldNameTranslators.java

```diff
@@ -83,8 +115,15 @@ static class PrefixFieldNameTranslator extends FieldNameTranslator {
         }
 
         @Override
-        boolean supports(String fieldName) {
-            return fieldName.startsWith(prefix);
+        boolean supports(String fieldNamePrefix) {
+            // a pattern can generally match a prefix in multiple ways
+            // moreover, it's not possible to iterate the concrete fields matching the prefix
+            if (Regex.isSimpleMatchPattern(fieldNamePrefix)) {
+                // this means `metadata.*` and `metadata.x*` are expanded to the empty list
+                // rather than be replaced with `metadata_flattened.*` and `metadata_flattened.x*`
```

Comment:
I found this a little confusing. Sorry if my question doesn't make sense! 

So, this means that we don't support wildcards for metadata attribute field selection, right? It has to be the exact name of the attribute to match? So we can't have `metadata.something_a` and `metadata.something_b` and select them for search with `"fields": ["metadata.something*"]`. Is it possible to document that part, so the part where the result of this `return false` is that we can't match on a simple match pattern when selecting metadata attributes, it's either all attributes or a list of specific attributes.

Reply:
It's indeed confusing. I'll jot something in the docs to explain it.

>So, this means that we don't support wildcards for metadata attribute field selection, right?

That's exactly right!

The root limitation actually comes from the way the [flattened field type](https://www.elastic.co/guide/en/elasticsearch/reference/current/flattened.html) works (the API key `metadata_flattened` field is mapped as such).
As you say, it's not possible to access a subset of the `metadata_flattened.*` fields via wildcards. It's either all of them, via `metadata_flattened`, or specific (non-wildcard) ones in a list.

Because the limitation comes from the Simple Query String processing, we could drop the processing here, i.e. which prohibits the `metadata.x*` -> `metadata_flattened.x*` translation, and it would still work identically (because `metadata_flattened.x*` is going to be implicitly ignored). But I think it's clearer to keep it (and test for it).

--------

Interestingly, if the sub-field is a wildcard, e.g. `metadata_flattened.wild*` exists as a field, then this field cannot be accessed in the `field` parameter of the `simple_query_string`.

Reply:
Pushed [18e9993](https://github.com/elastic/elasticsearch/pull/104132/commits/18e999342ef3b81f26c0bd375a0a7a42129709eb) to explain the above in the docs proper.